query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresBasedRequestHandler.java

```diff
@@ -0,0 +1,634 @@
+  private int slowQueryThreshold = DEFAULT_SLOW_QUERY_THRESHOLD_MS;
+
+  PostgresBasedRequestHandler(String name, Config config) {
+    this(name, config, new DefaultResultSetTypePredicateProvider(), PostgresClientFactory.get());
+  }
+
+  PostgresBasedRequestHandler(
+      String name,
+      Config config,
+      ResultSetTypePredicateProvider resultSetTypePredicateProvider,
+      PostgresClientFactory postgresClientFactory) {
+    this.name = name;
+    this.resultSetTypePredicateProvider = resultSetTypePredicateProvider;
+    this.postgresClientFactory = postgresClientFactory;
+    this.processConfig(config);
+  }
+
+  private void initMetrics() {
+    // Registry the latency metric with handler as a tag.
+    this.postgresQueryExecutionTimer =
+        PlatformMetricsRegistry.registerTimer(
+            "postgres.query.latency", Map.of("handler", name), true);
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public Optional<String> getTimeFilterColumn() {
+    return this.startTimeAttributeName;
+  }
+
+  private void processConfig(Config config) {
+
+    if (!config.hasPath(TENANT_COLUMN_NAME_CONFIG_KEY)) {
+      throw new RuntimeException(
+          TENANT_COLUMN_NAME_CONFIG_KEY + " is not defined in the " + name + " request handler.");
+    }
+
+    String tenantColumnName = config.getString(TENANT_COLUMN_NAME_CONFIG_KEY);
+    String countColumnName =
+        config.hasPath(COUNT_COLUMN_NAME_CONFIG_KEY)
+            ? config.getString(COUNT_COLUMN_NAME_CONFIG_KEY)
+            : null;
+
+    this.tableDefinition =
+        TableDefinition.parse(
+            config.getConfig(TABLE_DEFINITION_CONFIG_KEY), tenantColumnName, countColumnName);
+
+    this.startTimeAttributeName =
+        config.hasPath(START_TIME_ATTRIBUTE_NAME_CONFIG_KEY)
+            ? Optional.of(config.getString(START_TIME_ATTRIBUTE_NAME_CONFIG_KEY))
+            : Optional.empty();
+
+    this.request2PostgresSqlConverter =
+        new QueryRequestToPostgresSQLConverter(
+            tableDefinition,
+            new PostgresFunctionConverter(
+                tableDefinition, new PostgresFunctionConverterConfig(config)));
+
+    if (config.hasPath(SLOW_QUERY_THRESHOLD_MS_CONFIG)) {
+      this.slowQueryThreshold = config.getInt(SLOW_QUERY_THRESHOLD_MS_CONFIG);
+    }
+    LOG.info(
+        "Using {}ms as the threshold for logging slow queries of handler: {}",
+        slowQueryThreshold,
+        name);
+
+    initMetrics();
+  }
+
+  /**
+   * Returns a QueryCost that is an indication of whether the given query can be handled by this
+   * handler and if so, how costly is it to handle that query.
+   *
+   * <p>A query can usually be handled by Postgres handler if the Postgres view of this handler has
+   * all the columns that are referenced in the incoming query. If the Postgres view is a filtered
+   * view on some view column filters, the incoming query has to have those filters to match the
```

Comment:
> If the Postgres view is a filtered view on some view column filters, the incoming query has to have those filters to match the view.

Didn't get fully. It should be all referenced columns including `selection, where, group_by, order_by, etc`, right?

Reply:
Just saw `viewDefinitionSupportsFilter`, I think, it is for additional configuration support.